src/common/packing.jl

```diff
@@ -0,0 +1,25 @@
+# Packing routines used in direct_minimization and newton algorithms.
+# They pack / unpack sets of planewaves to make them compatible to be used in
+# algorithms from KrylovKit or Optim libraries
+
+pack(φ) = vcat(Base.vec.(φ)...) # TODO as an optimization, do that lazily? See LazyArrays
+function unpacking(φ) where T
+
+    Nk = length(φ)
+
+    lengths = [length(φ[ik]) for ik = 1:Nk]
+    starts = copy(lengths)
+    starts[1] = 1
+    for ik = 1:Nk-1
+        starts[ik+1] = starts[ik] + lengths[ik]
+    end
```

Comment:
Use `cumsum`

Reply:
Hum, not sure that it makes it easier no ? I don't really see how cumsum can help here :thinking:

Reply:
Well you have the ranges defined by the starts. With cumsum(lengths) you had them defined by the ends, which you could use in your definition below and you avoid writing out the loops.